ssl/ssl_lib.c

```diff
@@ -7211,3 +7264,199 @@ int SSL_stream_conclude(SSL *ssl, uint64_t flags)
     return 0;
 #endif
 }
+
+int SSL_add_expected_rpk(SSL *s, EVP_PKEY *rpk, uint8_t mtype)
```

Comment:
It occurs to me that perhaps this function should simply drop the `mtype` argument, and always behave as if `OSSL_DANETLS_MATCHING_FULL` was specified. After all, the caller already has the full key "in hand", no need to digest it for publication in DNS, ... There's not much point in making the synthetic TLSA record a digest...

What do you think?  If the user wants to use digested TLSA records (perhaps they only a key digest) they can use the TLSA API (now that the code point macros are public).

Reply:
Specifically, I have a proposed commit to make it so: https://github.com/vdukhovni/openssl/commit/a58b92fbe10e8bb9748ec75d910020b0ec9dd157

Reply:
And separately, also https://github.com/vdukhovni/openssl/commit/56050a712f47152ba6d6a7c4e05cc45856ecb17f

Reply:
I would prefer to keep it as-is. It's an ease-of-use thing. And how far does one go, as the next step is to get rid of the function completely; which I think is necessary for ease-of-use.

Reply:
> I would prefer to keep it as-is. It's an ease-of-use thing. And how far does one go, as the next step is to get rid of the function completely; which I think is necessary for ease-of-use.

I agree the function is useful when you already have the full key in hand, but I don't agree that having to specify the matching type aids usability. I can't think of any reason to prefer to match a digest when you have the full key in hand, and can match the key directly. The various matching types in DANE are there to optimise publication via DNS, where full RSA keys are needless bloat.

So the proposed change is by no means intended to strip away the function, just make it *more* usable, by "unasking" the choice of `mtype`.

Reply:
Yeah, I just have to find the time